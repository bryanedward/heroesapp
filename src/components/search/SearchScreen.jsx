import React, { useMemo, useState } from 'react'
import { Navigate, useLocation, useNavigate } from 'react-router-dom'
import { getByNameHero } from '../../selectors/getByNameHero';
import { HeroCard } from '../hero/HeroCard';
const queryString = require('query-string');

export const SearchScreen = () => {

  const navigate = useNavigate()
  const location = useLocation()
  const [value, setvalue] = useState('')
  const { q = '' } = queryString.parse(location.search)
  const getAllHero = useMemo(() => getByNameHero(q),[q]) 

  const btnSearch = (e) => {
    e.preventDefault()
    navigate(`?q=${value}`)
  }

  const valuesInput = (e) => {
    setvalue(e.target.value)
  }

  return (
    <div className='d-flex justify-content-around'>
      <div>
        <form onSubmit={btnSearch}>
          <p>Buscar</p>
          <input type="text" onChange={valuesInput} />
          <button className='btn btn-success' onClick={btnSearch}>
            Buscar</button>
        </form>
      </div>
      <div>

        {
          (!getAllHero) ? <p>no hay datos</p> :
            getAllHero.map((item, index) => (
              <HeroCard key={index} {...item} />
            ))
        }
      </div>
    </div>
  )
}
