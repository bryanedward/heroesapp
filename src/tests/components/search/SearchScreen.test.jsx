import  {mount} from 'enzyme'
import { MemoryRouter, useNavigate } from 'react-router-dom'
import { SearchScreen } from '../../../components/search/SearchScreen'

const mockNavigate = jest.fn()

jest.mock('react-router-dom',() => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate:() => mockNavigate

}))

describe('comprobacion del componente serach',() => {



    test('validar el componente search', () => { 
        const wrapper = mount(<MemoryRouter initialEntries={['/search']}>
            <SearchScreen />
        </MemoryRouter>)
        expect(wrapper).toMatchSnapshot()
     })

     test('validar el search del url y input', () => { 
         const wrapper = mount(<MemoryRouter initialEntries={['/search?q=batman']}>
             <SearchScreen/>
         </MemoryRouter>)
         expect(wrapper.find('input').prop('value')).toBe('batman')
      })

      test('validar el submit del formulario', () => { 
          const wrapper = mount(<MemoryRouter initialEntries={[
              '/search?q=batman'
          ]}>
              <SearchScreen/>
          </MemoryRouter>)
          wrapper.find('input').simulate('change',{
              target:{
                  value:'batman',
                  name:'value'
              }
          })
          wrapper.find('form').prop('onSubmit')({
              preventDefault(){}
          })

          expect(mockNavigate).toBeCalledWith('?q=batman')

       })
})