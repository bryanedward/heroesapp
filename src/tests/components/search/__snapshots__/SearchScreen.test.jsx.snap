// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` queryparamts 1`] = `
<div
  className="d-flex justify-content-around"
>
  <div>
    <form
      onSubmit={[Function]}
    >
      <p>
        Buscar
      </p>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="btn btn-success"
        onClick={[Function]}
      >
        Buscar
      </button>
    </form>
  </div>
  <div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-batman.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Batman
      </p>
      <a
        href="/heroe/dc-batman"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
  </div>
</div>
`;

exports[` should first 1`] = `
<div
  className="d-flex justify-content-around"
>
  <div>
    <form
      onSubmit={[Function]}
    >
      <p>
        Buscar
      </p>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="btn btn-success"
        onClick={[Function]}
      >
        Buscar
      </button>
    </form>
  </div>
  <div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-batman.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Batman
      </p>
      <a
        href="/heroe/dc-batman"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-superman.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Superman
      </p>
      <a
        href="/heroe/dc-superman"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-flash.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Flash
      </p>
      <p
        className="text-muted"
      >
        Jay Garrick, Barry Allen, Wally West, Bart Allen
      </p>
      <a
        href="/heroe/dc-flash"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-green.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Green Lantern
      </p>
      <p
        className="text-muted"
      >
        Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
      </p>
      <a
        href="/heroe/dc-green"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-arrow.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Green Arrow
      </p>
      <a
        href="/heroe/dc-arrow"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-wonder.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Wonder Woman
      </p>
      <a
        href="/heroe/dc-wonder"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-martian.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Martian Manhunter
      </p>
      <p
        className="text-muted"
      >
        Martian Manhunter
      </p>
      <a
        href="/heroe/dc-martian"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-robin.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Robin/Nightwing
      </p>
      <a
        href="/heroe/dc-robin"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-blue.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Blue Beetle
      </p>
      <p
        className="text-muted"
      >
        Dan Garret, Ted Kord, Jaime Reyes
      </p>
      <a
        href="/heroe/dc-blue"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-black.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Black Canary
      </p>
      <p
        className="text-muted"
      >
        Dinah Drake, Dinah Lance
      </p>
      <a
        href="/heroe/dc-black"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-spider.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Spider Man
      </p>
      <a
        href="/heroe/marvel-spider"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-captain.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Captain America
      </p>
      <a
        href="/heroe/marvel-captain"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-iron.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Iron Man
      </p>
      <a
        href="/heroe/marvel-iron"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-thor.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Thor
      </p>
      <a
        href="/heroe/marvel-thor"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-hulk.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Hulk
      </p>
      <a
        href="/heroe/marvel-hulk"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-wolverine.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Wolverine
      </p>
      <a
        href="/heroe/marvel-wolverine"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-daredevil.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Daredevil
      </p>
      <a
        href="/heroe/marvel-daredevil"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-hawkeye.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Hawkeye
      </p>
      <a
        href="/heroe/marvel-hawkeye"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-cyclops.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Cyclops
      </p>
      <a
        href="/heroe/marvel-cyclops"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-silver.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Silver Surfer
      </p>
      <a
        href="/heroe/marvel-silver"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
  </div>
</div>
`;

exports[`comprobacion del componente serach should first 1`] = `
<div
  className="d-flex justify-content-around"
>
  <div>
    <form
      onSubmit={[Function]}
    >
      <p>
        Buscar
      </p>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="btn btn-success"
        onClick={[Function]}
      >
        Buscar
      </button>
    </form>
  </div>
  <div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-batman.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Batman
      </p>
      <a
        href="/heroe/dc-batman"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-superman.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Superman
      </p>
      <a
        href="/heroe/dc-superman"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-flash.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Flash
      </p>
      <p
        className="text-muted"
      >
        Jay Garrick, Barry Allen, Wally West, Bart Allen
      </p>
      <a
        href="/heroe/dc-flash"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-green.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Green Lantern
      </p>
      <p
        className="text-muted"
      >
        Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
      </p>
      <a
        href="/heroe/dc-green"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-arrow.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Green Arrow
      </p>
      <a
        href="/heroe/dc-arrow"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-wonder.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Wonder Woman
      </p>
      <a
        href="/heroe/dc-wonder"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-martian.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Martian Manhunter
      </p>
      <p
        className="text-muted"
      >
        Martian Manhunter
      </p>
      <a
        href="/heroe/dc-martian"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-robin.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Robin/Nightwing
      </p>
      <a
        href="/heroe/dc-robin"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-blue.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Blue Beetle
      </p>
      <p
        className="text-muted"
      >
        Dan Garret, Ted Kord, Jaime Reyes
      </p>
      <a
        href="/heroe/dc-blue"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-black.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Black Canary
      </p>
      <p
        className="text-muted"
      >
        Dinah Drake, Dinah Lance
      </p>
      <a
        href="/heroe/dc-black"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-spider.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Spider Man
      </p>
      <a
        href="/heroe/marvel-spider"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-captain.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Captain America
      </p>
      <a
        href="/heroe/marvel-captain"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-iron.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Iron Man
      </p>
      <a
        href="/heroe/marvel-iron"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-thor.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Thor
      </p>
      <a
        href="/heroe/marvel-thor"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-hulk.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Hulk
      </p>
      <a
        href="/heroe/marvel-hulk"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-wolverine.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Wolverine
      </p>
      <a
        href="/heroe/marvel-wolverine"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-daredevil.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Daredevil
      </p>
      <a
        href="/heroe/marvel-daredevil"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-hawkeye.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Hawkeye
      </p>
      <a
        href="/heroe/marvel-hawkeye"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-cyclops.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Cyclops
      </p>
      <a
        href="/heroe/marvel-cyclops"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-silver.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Silver Surfer
      </p>
      <a
        href="/heroe/marvel-silver"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
  </div>
</div>
`;

exports[`comprobacion del componente serach validar el componente search 1`] = `
<div
  className="d-flex justify-content-around"
>
  <div>
    <form
      onSubmit={[Function]}
    >
      <p>
        Buscar
      </p>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="btn btn-success"
        onClick={[Function]}
      >
        Buscar
      </button>
    </form>
  </div>
  <div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-batman.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Batman
      </p>
      <a
        href="/heroe/dc-batman"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-superman.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Superman
      </p>
      <a
        href="/heroe/dc-superman"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-flash.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Flash
      </p>
      <p
        className="text-muted"
      >
        Jay Garrick, Barry Allen, Wally West, Bart Allen
      </p>
      <a
        href="/heroe/dc-flash"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-green.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Green Lantern
      </p>
      <p
        className="text-muted"
      >
        Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
      </p>
      <a
        href="/heroe/dc-green"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-arrow.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Green Arrow
      </p>
      <a
        href="/heroe/dc-arrow"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-wonder.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Wonder Woman
      </p>
      <a
        href="/heroe/dc-wonder"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-martian.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Martian Manhunter
      </p>
      <p
        className="text-muted"
      >
        Martian Manhunter
      </p>
      <a
        href="/heroe/dc-martian"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-robin.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Robin/Nightwing
      </p>
      <a
        href="/heroe/dc-robin"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-blue.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Blue Beetle
      </p>
      <p
        className="text-muted"
      >
        Dan Garret, Ted Kord, Jaime Reyes
      </p>
      <a
        href="/heroe/dc-blue"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/dc-black.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Black Canary
      </p>
      <p
        className="text-muted"
      >
        Dinah Drake, Dinah Lance
      </p>
      <a
        href="/heroe/dc-black"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-spider.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Spider Man
      </p>
      <a
        href="/heroe/marvel-spider"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-captain.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Captain America
      </p>
      <a
        href="/heroe/marvel-captain"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-iron.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Iron Man
      </p>
      <a
        href="/heroe/marvel-iron"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-thor.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Thor
      </p>
      <a
        href="/heroe/marvel-thor"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-hulk.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Hulk
      </p>
      <a
        href="/heroe/marvel-hulk"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-wolverine.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Wolverine
      </p>
      <a
        href="/heroe/marvel-wolverine"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-daredevil.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Daredevil
      </p>
      <a
        href="/heroe/marvel-daredevil"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-hawkeye.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Hawkeye
      </p>
      <a
        href="/heroe/marvel-hawkeye"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-cyclops.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Cyclops
      </p>
      <a
        href="/heroe/marvel-cyclops"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
    <div
      className="card"
      style={
        Object {
          "width": "13pc",
        }
      }
    >
      <img
        alt=""
        src="assets/marvel-silver.jpg"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
      <p>
        Silver Surfer
      </p>
      <a
        href="/heroe/marvel-silver"
        onClick={[Function]}
      >
        ver mas...
      </a>
    </div>
  </div>
</div>
`;
